zerver/data_import/slack.py

```diff
@@ -450,29 +452,30 @@ def convert_slack_workspace_messages(slack_data_dir: str, users: List[ZerverFiel
     """
     all_messages = get_all_messages(slack_data_dir, added_channels)
 
-    # we sort the messages according to the timestamp to show messages with
-    # the proper date order
-    all_messages = sorted(all_messages, key=lambda message: message['ts'])
-
     logging.info('######### IMPORTING MESSAGES STARTED #########\n')
 
     total_reactions = []  # type: List[ZerverFieldsT]
     total_attachments = []  # type: List[ZerverFieldsT]
     total_uploads = []  # type: List[ZerverFieldsT]
 
     # The messages are stored in batches
-    low_index = 0
-    upper_index = low_index + chunk_size
     dump_file_id = 1
 
     subscriber_map = make_subscriber_map(
         zerver_subscription=realm['zerver_subscription'],
     )
 
     while True:
-        message_data = all_messages[low_index:upper_index]
+        message_data = []
+        _counter = 0
+        for msg in all_messages:
```

Comment:
Ok, I am not sure why the loop causes `channel_message_to_zerver_message` to be iterated one step more than it should be.

Reply:
Fixed with casting `iter` to mock `get_all_messages`.